### ISSUE TITLE
cquery --output=files assume convenience symlinks are enabled

### ISSUE THREAD

**Issue by stevebarrau** [bazelbuild/bazel#16277]
### Description of the bug:

Bazel cquery `--output=files` assumes the convenience symlinks are enabled and outputs files with them. It does not take the `--experimental_convenience_symlinks=ignore` option into account.

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```console
# in the bazel repo, on master (390d212d76e91f413f143b53a997b0bf846c453a)
$ bazel cquery --output=files //examples/cpp:hello-world 2>/dev/null
bazel-out/darwin-fastbuild/bin/examples/cpp/hello-world
# this is expected
$ bazel cquery --experimental_convenience_symlinks=ignore --output=files //examples/cpp:hello-world 2>/dev/null
bazel-out/darwin-fastbuild/bin/examples/cpp/hello-world
# this I would expect to output the output_base instead
```

### Which operating system are you running Bazel on?

macOS 12.4

### What is the output of `bazel info release`?

release 5.3.0

### What's the output of `git remote get-url origin; git rev-parse master; git rev-parse HEAD` ?

```text
git@github.com:bazelbuild/bazel.git
390d212d76e91f413f143b53a997b0bf846c453a
390d212d76e91f413f143b53a997b0bf846c453a
```

**Comment by fmeum** [comment]
Valid issue, I didn't think of that flag when implementing the feature. I realistically won't be able to work on this soon, but would offer assistance to anyone interested in working on a fix.

**Comment by brentleyjones** [comment]
I think the paths aren't based on convenience symlinks, but instead relative to `execution_root`

**Comment by fmeum** [comment]
> I think the paths aren't based on convenience symlinks, but instead relative to `execution_root`

Yes, they are relative to the exec root. Even with convenience symlinks disabled, appending them to `$(bazel info execution_root)` should make them resolve. But of course providing absolute paths if the symlinks aren't created would be more user-friendly.

@gregestren What do you think, should `--output=files` always return exec paths for consistency or ensure that its output paths always resolve relative to the workspace root (i.e., are absolute when not using the convenience symlinks)?

**Comment by gregestren** [comment]
It's not obvious to me the current behavior is bad, for the reasons @brentleyjones explains. 

Just to sanity check, @stevebarrau  the output should still be valid even without convenience symlinks. `bazel-out/darwin-fastbuild/bin/examples/cpp/hello-world` doesn't rely on the convenience symlinks. `bazel-out` *is* a symlink, but a different kind. 

Is there a specific scenario you're finding breaking from this behavior? Like building from a certain directory where `bazel-out/...` doesn't resolve easily for you?

**Comment by stevebarrau** [comment]
My use case comes back to a few tickets I have seen around bazel TUI verbosity and stdout. I would like bazel to print to stdout where a binary is built. 

I want to replace my current use of “bazel run —run_undo=echo //:foo” with a build followed by a cquery —output=files. The former provide an absolute path, the later provide a path apparently rooted in the execution_root.

I thought this was about the convenience symlinks but it might actually just be a documentation improvement request.

**Comment by ankit-agarwal1999** [comment]
Reviving this for a relevant question: 

Is there a way to have bazel build not output the path in bazel-bin but output the path in bazel-out. The reason being, folks will run a command to build something, see the output under bazel-bin, and then run another command (which could invalidate bazel-bin) and expect the output to be stable. 

bazel-out is significantly more stable than bazel-bin so wondering if I can change bazel's output to be the bazel-out vs bazel-bin
